source/BuffPanelSdk/Client.cpp

```diff
@@ -83,17 +107,104 @@ void BuffPanel::Client::track(
 		std::istream& responsePayloadStream = session.receiveResponse(response);
 		Poco::JSON::Parser jsonParser;
 		jsonParser.parse(responsePayloadStream);
-				
+
 		// Validate the response.
 		if (response.getStatus() == Poco::Net::HTTPResponse::HTTPStatus::HTTP_OK) {
 			// Notify the callback object that the event was successfully processed.
 			callback.success();
-		} else {
+		}
+		else {
 			// Notify the callback object that the server failed to process the request.
 			callback.error("An invalid response was recieved from the server.");
-		}		
-	} catch (Poco::Exception& exception) {
+		}
+	}
+	catch (Poco::Exception& exception) {
 		// Notify the callback object that an error occured while attempting to communicate with the server.
 		callback.error("An error occured while attempting to communicate with the server: " + exception.displayText());
 	}
 }
+
+
+std::string BuffPanel::Client::generateUuid()
+{
+	return Poco::UUIDGenerator().createRandom().toString();
+}
+
+std::string BuffPanel::Client::getUuidPersistPath()
+{
+	Poco::Path uuidPersistPath;
+#ifdef _WIN32 
+	wchar_t wpath[MAX_PATH];
+	HRESULT rc = SHGetFolderPathW(
+		nullptr,
+		CSIDL_LOCAL_APPDATA | CSIDL_FLAG_CREATE,
+		nullptr, SHGFP_TYPE_CURRENT,
+		wpath);
+	if (SUCCEEDED(rc))
+	{
+		std::string localAppData;
+		Poco::UnicodeConverter::toUTF8(wpath, localAppData);
+		uuidPersistPath = Poco::Path(localAppData);
+		uuidPersistPath.makeDirectory();
+		uuidPersistPath.pushDirectory("BuffPanel");
+	}
+#elif __APPLE__
+	uuidPersistPath = Poco::Path(Poco::Path::home());
+	uuidPersistPath.pushDirectory(".local");
+	uuidPersistPath.pushDirectory("BuffPanel");
+#elif __linux__
+	uuidPersistPath = Poco::Path(Poco::Path::home());
+	uuidPersistPath.pushDirectory(".local");
+	uuidPersistPath.pushDirectory("BuffPanel");
+#endif
+	return uuidPersistPath.toString();
+}
+
+std::string BuffPanel::Client::readSavedUuid(const std::string& path)
+{
+	if (!Poco::File(path).exists() || !Poco::File(path).canRead())
+	{
+		return std::string("");
+	}
+	std::string result;
+	Poco::FileInputStream inStream(path);
+	Poco::StreamCopier::copyToString(inStream, result);
+	inStream.close();
+
+	Poco::UUID pocoUuid;
+	if (!pocoUuid.tryParse(result))
+		return std::string("");
+
+	return result;
+}
+
+void BuffPanel::Client::saveUuid(const std::string& filePath, const std::string& folderPath, const std::string& uuid)
+{
+	Poco::File buffPanelDir(folderPath);
+	buffPanelDir.createDirectory();
+
+	Poco::File buffPanelFile(filePath);
+
+	if (buffPanelFile.exists())
+	{
+		buffPanelFile.remove();
+	}
+	buffPanelFile.createFile();
+	Poco::FileOutputStream outStream(buffPanelFile.path());
+	outStream << uuid << std::endl;
+	outStream.close();
+
+}
+
+std::string BuffPanel::Client::getPlayerToken(const std::string& gameToken)
+{
+	const std::string folderPath = getUuidPersistPath();
+	const std::string filePath = folderPath + "uuid_" + gameToken;
+	std::string uuid = readSavedUuid(filePath);
+	if (uuid.empty())
```

Comment:
or `uuid == "anonymous"`

Reply:
actually it turns out, this is the right implementation... I had little bit of chaos in this, but if uuid is "anonymous", then we don't want to generate and persist UUID (because of some permissions problem etc..)